Tests.Unit.Core.ApplicationServices/ApplicationServices/Users/UserWriteServiceTest.cs

```diff
@@ -0,0 +1,157 @@
+            _transactionManagerMock = new Mock<ITransactionManager>();
+
+            _sut = new UserWriteService(_userServiceMock.Object, 
+                _entityIdentityResolverMock.Object,
+                _organizationRightsServiceMock.Object, 
+                _transactionManagerMock.Object);
+        }
+
+        [Fact]
+        public void Can_Create_User()
+        {
+            //Arrange
+            var createParams = SetupUserParameters();
+            var orgUuid = A<Guid>();
+            var orgId = A<int>();
+            var transaction = ExpectTransactionBegins();
+
+            AssertResolveIdReturns(orgUuid, orgId);
+            AssertAddUserReturns(createParams.User, createParams.SendMailOnCreation, orgId);
+            foreach (var organizationRole in createParams.Roles)
+            {
+                AssertAddRoleReturns(organizationRole, orgId, createParams.User.Id, Result<OrganizationRight, OperationFailure>.Success(It.IsAny<OrganizationRight>()));
+            }
+
+            //Act
+            var result = _sut.Create(orgUuid, createParams);
+
+            //Assert
+            Assert.True(result.Ok);
+            transaction.Verify(x => x.Commit(), Times.Once);
+        }
+
+        [Fact]
+        public void Create_Fails_If_Add_Role_Fails()
+        {
+            //Arrange
+            var createParams = SetupUserParameters();
+            var orgUuid = A<Guid>();
+            var orgId = A<int>();
+            var transaction = ExpectTransactionBegins();
+            var error = A<OperationFailure>();
+
+            AssertResolveIdReturns(orgUuid, orgId);
+            AssertAddUserReturns(createParams.User, createParams.SendMailOnCreation, orgId);
+            foreach (var organizationRole in createParams.Roles)
+            {
+                AssertAddRoleReturns(organizationRole, orgId, createParams.User.Id, error);
+            }
+
+            //Act
+            var result = _sut.Create(orgUuid, createParams);
+
+            //Assert
+            Assert.True(result.Failed);
+            Assert.Equal(error, result.Error.FailureType);
+            Assert.True(result.Error.Message.HasValue);
+            Assert.True(result.Error.Message.Value.Contains("Failed to assign role"));
+            transaction.Verify(x => x.Rollback(), Times.Once);
+        }
+
+        [Fact]
+        public void Create_Fails_If_OrgId_Is_Not_Found()
+        {
+            //Arrange
+            var createParams = SetupUserParameters();
+            var orgUuid = A<Guid>();
+
+            AssertResolveIdReturns(orgUuid, Maybe<int>.None);
+
+            //Act
+            var result = _sut.Create(orgUuid, createParams);
+
+            //Assert
+            Assert.True(result.Failed);
+            Assert.Equal(OperationFailure.NotFound, result.Error.FailureType);
+        }
+
+        private void AssertAddUserReturns(User user, bool sendMailOnCreation, int orgId)
```

Comment:
Shouldn' these 3 methods be called "Expect...Someting" instead of "Assert...Something"? When i read them, i thought they were asserting some properties, and not being used for setup.

Reply:
@JacobBorchMadsen In a way you're right, but we're actually Asserting that they are going to return this value. if we specify it here we tell the Mocked Service to "assert" that the given method is returning the value we want it to return

Reply:
But I also see the "expect" name making sense. I think I like it more